rust/src/lib.rs

```diff
@@ -184,24 +275,20 @@ pub fn add_to_linker<T: 'static>(
 ) -> Result<EventChannel> {
     let (ctx, events_tx, events_rx) = InstrumentationContext::new(id);
 
-    // Build up a table of function id to name mapping
-    let mut function_names = FunctionNames::new();
-    let parser = wasmparser::Parser::new(0);
-    for payload in parser.parse_all(data) {
-        if let wasmparser::Payload::CustomSection(custom) = payload? {
-            if custom.name() == "name" {
-                let name_reader =
-                    wasmparser::NameSectionReader::new(custom.data(), custom.data_offset());
-                for x in name_reader.into_iter() {
-                    if let wasmparser::Name::Function(f) = x? {
-                        for k in f.into_iter() {
-                            let k = k?;
-                            function_names.insert(k.index, k.name.to_string());
-                        }
-                    }
-                }
-                continue;
-            }
+    // load the static wasm-instr info
+    let wasm_instr_info = WasmInstrInfo::new(data)?;
+
+    // For now tolerate version info not being present
+    if wasm_instr_info.maj_version.is_some() || wasm_instr_info.min_version.is_some() {
+        let maj_num = wasm_instr_info.maj_version.unwrap();
+        let min_num = wasm_instr_info.min_version.unwrap();
```

Comment:
The current logic could lead to an unwrap failing on the next line if one of these is `None` - this does seem really unlikely, but still worth fixing.

```suggestion
    if let WasmInstrInfo {maj_version: Some(maj_num), min_version: Some(min_num), ..} = &wasm_instr_info {
```